h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -137,18 +136,48 @@ class SearchBarController extends Controller {
     };
 
     /**
-     * Updates the hidden input field with the consolidated
-     * search terms from the lozenges.
+     * Insert a hidden <input> into the search <form>.
+     *
+     * The value of the hidden <input> is a search string constructed from
+     * the lozenges plus any text currently in the visible <input>.
+     *
+     * The name="q" attribute is moved from the visible <input> on to the
+     * hidden <input> so that when the <form> is submitted it's the value of
+     * the _hidden_ input, not the visible one, that is submitted as the
+     * q parameter.
+     *
      */
-    var updateHiddenInputValue = () => {
-      var newValue = '';
-      Array.from(this._lozengeContainer.querySelectorAll('.js-lozenge__content'), function(loz) {
-        newValue = newValue + loz.textContent + ' ';
+    var insertHiddenInput = () => {
+      var hiddenInput = document.createElement('input');
+      hiddenInput.type = 'hidden';
+
+      Array.from(this._lozengeContainer.querySelectorAll('.js-lozenge__content')).forEach((loz) => {
```

Comment:
Is that right? I wasn't getting a linter warning. Looked through the airbnb guide, closest I could find was [If your function takes a single argument *and doesn’t use braces*, omit the parentheses. Otherwise, always include parentheses around arguments. ](https://github.com/airbnb/javascript#arrows--one-arg-parens). I think this would come under the otherwise because it does use braces. And indeed if I remove the braces then, if I'm using eslint-config-airbnb, eslint complains _Expected parens around arrow function having a body with curly braces_.

By the way should we change [our eslint config](https://github.com/hypothesis/h/blob/master/.eslintrc) to `"extends": "airbnb"` or make eslint-config-hypothesis extend airbnb? (And then also add instructions, or link to instructions, for installing eslint-config-airbnb as it's a bit tricky.) I can send a PR for that if you want. eslint-config-airbnb has **176** warnings about `search-bar-controller.js` which is a _new_ file!